sklearnex/neighbors/common.py

```diff
@@ -262,64 +272,12 @@ def _onedal_supported(self, device, method_name, *data):
             return patching_status
         raise RuntimeError(f"Unknown method {method_name} in {class_name}")
 
-    def _kneighbors_dispatch(
-        self, branches, X=None, n_neighbors=None, return_distance=True
-    ):
-        if n_neighbors is None:
-            n_neighbors = self.n_neighbors
-        elif n_neighbors <= 0:
-            raise ValueError("Expected n_neighbors > 0. Got %d" % n_neighbors)
-        else:
-            if not isinstance(n_neighbors, Integral):
-                raise TypeError(
-                    "n_neighbors does not take %s value, "
-                    "enter integer value" % type(n_neighbors)
+    def _fit_queue_check(self, X, queue=None):
+        if X is None and hasattr(self, "_fit_queue"):
+            if queue is None:
+                queue = self._fit_queue
+            elif queue != self._fit_queue:
+                raise RuntimeError(
+                    "Input data shall be located " "on single target device"
                 )
-
-        query_is_train = X is None
-        if query_is_train:
-            X = self._fit_X
-            n_neighbors += 1
-
-        results = dispatch(
-            self,
-            "kneighbors",
-            branches,
-            X,
-            n_neighbors,
-            return_distance,
-        )
-
-        if not query_is_train:
-            return results
-        # If the query data is the same as the indexed data, we would like
-        # to ignore the first nearest neighbor of every sample, i.e
-        # the sample itself.
-        if return_distance:
-            neigh_dist, neigh_ind = results
-        else:
-            neigh_ind = results
-
-        n_queries = X.shape[0] if hasattr(X, "shape") else len(X)
-        sample_range = np.arange(n_queries)[:, None]
-        sample_mask = neigh_ind != sample_range
-
-        # Corner case: When the number of duplicates are more
-        # than the number of neighbors, the first NN will not
-        # be the sample, but a duplicate.
-        # In that case mask the first duplicate.
-        dup_gr_nbrs = np.all(sample_mask, axis=1)
-        sample_mask[:, 0][dup_gr_nbrs] = False
-
-        neigh_ind = np.reshape(neigh_ind[sample_mask], (n_queries, n_neighbors - 1))
-
-        if return_distance:
-            neigh_dist = np.reshape(neigh_dist[sample_mask], (n_queries, n_neighbors - 1))
-            return neigh_dist, neigh_ind
-        return neigh_ind
-
-    def _onedal_gpu_supported(self, method_name, *data):
-        return self._onedal_supported("gpu", method_name, *data)
-
-    def _onedal_cpu_supported(self, method_name, *data):
-        return self._onedal_supported("cpu", method_name, *data)
+        return queue
```

Comment:
It is weird and unexpected that we have this checker here?
What is KNN case? Why we have this work around and can not to use common approach?